diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5347218..3d630f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,7 @@ add_library (utf8proc
   utf8proc.c
   utf8proc.h
 )
+add_library (utf8proc::utf8proc ALIAS utf8proc)

 # expose header path, for when this is part of a larger cmake project
 target_include_directories(utf8proc PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include>)
